python/ray/worker.py

```diff
@@ -2017,7 +2026,10 @@ def connect(info,
     assert not worker.connected, error_message
     assert worker.cached_functions_to_run is not None, error_message
     # Initialize some fields.
-    worker.worker_id = random_string()
+    if worker_id is None:
```

Comment:
We should do some type checking. I think we want to assert that it is a Ray ObjectID or that it is 20 bytes or something like that. All of that is a bit ugly unfortunately.